Load `babel-plugin-syntax-dynamic-import` plugin

npejo opened the pull request:
From Webpack2 we can use dynamic imports to asynchronously load modules in the project. Even Webpack will understands the `import()` statement, Babel will fail to parse this code if we don't install the appropriate plugins.
> Module build failed: SyntaxError: 'import' and 'export' may only appear at the top level

To avoid this issue, we can use the `babel-plugin-syntax-dynamic-imports`.
> This plugin only allows Babel to parse this syntax.
https://babeljs.io/docs/plugins/syntax-dynamic-import/

pirelenito commented on src/configure-webpack/loaders/javascript.js:
```diff
@@ -50,7 +50,8 @@ export default {
                   [require.resolve('babel-plugin-transform-react-jsx-source'), {}],
-                  [require.resolve('babel-plugin-transform-react-jsx-self'), {}]
+                  [require.resolve('babel-plugin-transform-react-jsx-self'), {}],
+                  [require.resolve('babel-plugin-syntax-dynamic-import'), {}]
```
This shouldn't be only available in development mode. You could use the dynamic import feature for production builds as well.

pirelenito replied:
It would be nice to have an integration test that tests this "dynamic import" syntax to see if it really works.

deepsweet replied:
Dynamic imports are handled natively by webpack >= 2. This plugin is a "syntax", not "transform", and it should be in dev only just for babel to not fail on parsing.

deepsweet commented:
Dynamic imports are handled natively by webpack >= 2. This plugin is a "syntax", not "transform", and it should be in dev only (and not for production) just for babel to not fail on parsing.

pirelenito commented:
@deepsweet you can still use dynamic imports in production code, and that code needs to be understood by Babel (even if Webpack supports it), correct? Or am I missing something?

deepsweet commented:
You are right.

npejo commented:
Initially I added the plugin under the `if (action === actions.DEVELOP)` clause, but the integration test fails to execute `actions.BUILD` with dynamic import in the code.
When I moved the plugin out of the `if`, to be always required, Babel is not complaining any more 🙂
